development/metamaskbot-build-announce.js

```diff
@@ -243,6 +248,60 @@ async function start() {
     console.log(`No results for ${summaryPlatform} found; skipping benchmark`);
   }
 
+  try {
+    const prBundleSizeStats = JSON.parse(
+      await fs.readFile(
+        path.resolve(
+          __dirname,
+          '..',
+          path.join('test-artifacts', 'chrome', 'mv3', 'bundle_size.json'),
+        ),
+        'utf-8',
+      ),
+    );
+
+    const devBundleSizeStats = await (
+      await fetch(bundleSizeDataUrl, {
+        method: 'GET',
+      })
+    ).json();
+
+    const prSizes = {
+      background: prBundleSizeStats.background.size,
+      ui: prBundleSizeStats.ui.size,
+      common: prBundleSizeStats.common.size,
+    };
+
+    const devSizes = Object.keys(prSizes).reduce((sizes, part) => {
+      sizes[part] = devBundleSizeStats[PARENT_COMMIT][part] || 0;
+      return sizes;
+    }, {});
+
+    const diffs = Object.keys(prSizes).reduce((output, part) => {
+      output[part] = prSizes[part] - devSizes[part];
+      return output;
+    }, {});
+
+    const sizeDiffRows = Object.keys(diffs).map(
+      (part) => `${part}: ${diffs[part]} bytes`,
+    );
+
+    const sizeDiffHiddenContent = `<ul>${sizeDiffRows
+      .map((row) => `<li>${row}</li>`)
+      .join('\n')}</ul>`;
+
+    const sizeDiffWarning =
+      diffs.background + diffs.common > 0
+        ? `🚨 Warning! Bundle size has increased!`
+        : `🚀 Bundle size reduced!`;
```

Comment:
Suggestion: Should we handle the case where the bundle size is unchanged as well? There are a fair number of PRs that don't affect the bundle at all (e.g. test changes), and seeing "Bundle size reduced!" for those would be confusing.

Reply:
Good point. I applied your suggestion and the diff warning should now be hidden if common + background diff is 0.

I would leave Diffs details section visible in any case as there could be a change in the UI size that the dev might be interested in checking.